### PR TITLE
#1684 PlainSearchReplaceTransformer case-insensitive mode

### LINKED ISSUE #1684: Ignore case sensitivity in PlainSearch&Replace
PlainSearchReplaceTransformer is always case-sensitive. In some cases it would be a good idea to have case-INsensitive variant. 

Let's add a checkbox switching this mode of behaviour.

### PR DESCRIPTION
**Pull request by jhorcicka**
Resolves #1684 

PlainSearchReplaceTransformer can work in case-insensitive mode.

case-insensitive

### CODE REVIEW

File: components/basic-transformers/src/main/java/org/datacleaner/beans/transform/PlainSearchReplaceTransformer.java [hunk 1]
```diff
@@ -66,6 +67,10 @@
     @Description("Replace the entire string when the search string is found.")
     boolean replaceEntireString = false;
 
+    @Configured(order = 4)
+    @Description("Case sensitivity will be ignored during searching and replacing. ")
+    boolean ignoreCaseSensitivity = false;
```
Comment (LosD) [inline components/basic-transformers/src/main/java/org/datacleaner/beans/transform/PlainSearchReplaceTransformer.java]: I'm not really sure if this is consistent with the rest of our settings, but I don't really like negative preferences.

It's not-not case sensitive... So it's case sensitive :)

File: components/basic-transformers/src/main/java/org/datacleaner/beans/transform/PlainSearchReplaceTransformer.java [hunk 2]
```diff
@@ -114,28 +119,42 @@ public OutputColumns getOutputColumns() {
...
+        return result;
+    }
+
+    private String replaceEntireString(final String value) {
+        for (final Entry<String, String> entry : replacements.entrySet()) {
+            final String replacementString = entry.getValue();
+            final String searchString = entry.getKey();
+
+            if (ignoreCaseSensitivity == true) {
+                if (value.toLowerCase().indexOf(searchString.toLowerCase()) != -1) {
```
Comment (LosD) [inline components/basic-transformers/src/main/java/org/datacleaner/beans/transform/PlainSearchReplaceTransformer.java]: Can be replaced with `contains()`

File: components/basic-transformers/src/main/java/org/datacleaner/beans/transform/PlainSearchReplaceTransformer.java [hunk 3]
```diff
@@ -114,28 +119,42 @@ public OutputColumns getOutputColumns() {
...
+                    return replacementString;
                 }
+            } else if (value.indexOf(searchString) != -1) {
```
Comment (LosD) [inline components/basic-transformers/src/main/java/org/datacleaner/beans/transform/PlainSearchReplaceTransformer.java]: Same here

**Comment by jhorcicka** [comment]
@LosD: updated.

**Comment by LosD** [comment]
LGTM, and test went fine. Let's merge when Travis is happy

**Pull request merged**